hq/app/AppComponents.scala

```diff
@@ -71,28 +74,48 @@ class AppComponents(context: Context)
   }
 
   private val snykConfig = Config.getSnykConfig(configuration)
-  private val googleAuthConfig = Config.googleSettings(httpConfiguration, configuration)
   private val ec2Clients = AWS.ec2Clients(configuration, availableRegions)
   private val cfnClients = AWS.cfnClients(configuration, availableRegions)
   private val taClients = AWS.taClients(configuration)
   private val s3Clients = AWS.s3Clients(configuration, availableRegions)
   private val iamClients = AWS.iamClients(configuration, availableRegions)
   private val efsClients = AWS.efsClients(configuration, availableRegions)
-  val securityCredentialsProvider =
-    new AWSCredentialsProviderChain(new ProfileCredentialsProvider("security"), DefaultAWSCredentialsProviderChain.getInstance())
+
+  private val securityCredentialsProvider = new AWSCredentialsProviderChain(
+    new ProfileCredentialsProvider("security"),
+    DefaultAWSCredentialsProviderChain.getInstance()
+  )
   private val securitySnsClient = AmazonSNSAsyncClientBuilder.standard()
     .withCredentials(securityCredentialsProvider)
     .withRegion(Config.region)
     .withClientConfiguration(new ClientConfiguration().withMaxConnections(10))
     .build()
-  private val securityCenterSettings = SecurityCenterSettings.newBuilder().setCredentialsProvider(Config.gcpCredentialsProvider(configuration)).build()
-  private val securityCenterClient = SecurityCenterClient.create(securityCenterSettings)
-  private val dynamoDbClient = AWS.dynamoDbClient(securityCredentialsProvider, Config.region, stage)
+  private val securitySsmClient = AWSSimpleSystemsManagementClientBuilder.standard()
+    .withCredentials(securityCredentialsProvider)
+    .withRegion(Config.region)
+    .build()
+  private val googleAuthConfig = Config.googleSettings(stage, stack, configuration, securitySsmClient)
+
+  private val securityDynamoDbClient = stage match {
+    case PROD =>
+      AmazonDynamoDBClientBuilder.standard()
+        .withCredentials(securityCredentialsProvider)
+        .withRegion(Config.region)
+        .build()
+    case DEV =>
+      AmazonDynamoDBClientBuilder.standard()
+        .withCredentials(securityCredentialsProvider)
+        .withEndpointConfiguration(new EndpointConfiguration("http://localhost:8000", Config.region.name))
+        .build()
+  }
```

Comment:
I've moved a few things around to avoid forward references and to group conceptually similar things together. I've also moved single-client initialisations out of the separate `AWS` object, as I don't think the indirection there was adding much value and there are already initialisations done inline in this file.